github/dependency_graph.go

```diff
@@ -28,12 +28,32 @@ type CreationInfo struct {
 type RepoDependencies struct {
 	SPDXID *string `json:"SPDXID,omitempty"`
 	// Package name
-	Name             *string `json:"name,omitempty"`
-	VersionInfo      *string `json:"versionInfo,omitempty"`
-	DownloadLocation *string `json:"downloadLocation,omitempty"`
-	FilesAnalyzed    *bool   `json:"filesAnalyzed,omitempty"`
-	LicenseConcluded *string `json:"licenseConcluded,omitempty"`
-	LicenseDeclared  *string `json:"licenseDeclared,omitempty"`
+	Name             *string        `json:"name,omitempty"`
+	VersionInfo      *string        `json:"versionInfo,omitempty"`
+	DownloadLocation *string        `json:"downloadLocation,omitempty"`
+	FilesAnalyzed    *bool          `json:"filesAnalyzed,omitempty"`
+	LicenseConcluded *string        `json:"licenseConcluded,omitempty"`
+	LicenseDeclared  *string        `json:"licenseDeclared,omitempty"`
+	ExternalRefs     []*ExternalRef `json:"externalRefs"`
+}
+
+// ExternalRef represents an external reference (e.g., PURL/SWID/CPE) for a package in the SBOM.
+type ExternalRef struct {
```

Comment:
Since our exported struct names are package-global, let's please add more details to the individual struct names to make them more identifiable-at-a-glance.

Question: are all 3 fields required? If not, you might consider adding `omitempty` to the optional fields and make those same fields pointers instead of values (i.e. `*string` instead of `string`).

```suggestion
// SBOMExternalRef represents an external reference (e.g., PURL/SWID/CPE) for a package in the SBOM.
type SBOMExternalRef struct {
```

Reply:
> Since our exported struct names are package-global, let's please add more details to the individual struct names to make them more identifiable-at-a-glance.

Here, do you mean to add details on stuffs that I added only, or beyond that also which have been already there. As I can see that exiting ones also lacks few details.

Reply:
> Question: are all 3 fields required? If not, you might consider adding omitempty to the optional fields and make those same fields pointers instead of values (i.e. *string instead of string).

Overall, the field `ExternalRefs` is optional, but if it is provided, then these 3 fields must be provided, they aren't optional.

Reply:
> > Here, do you mean to add details on stuffs that I added only, or beyond that also which have been already there. As I can see that exiting ones also lacks few details.
> 
> Sorry I wasn't very clear... here I was just trying to explain the reasoning for my suggestion of changing "ExternalRef" (which is rather vague) to "SBOMExternalRef" so that it is more descriptive and more tightly-coupled to the SBOM parts of the API.

Oh, here you mean to add detail on struct name so that it looks identifiable-at-a-glance. Btw, I added comments describing about each struct fields. Should i remove them, or let it be added ?